draft-ietf-dnsop-glue-is-not-optional.md

```diff
@@ -166,50 +179,60 @@ coding = "utf-8"
    dhhs.gov.               3600    IN DS      635 8 1 ...
    dhhs.gov.               3600    IN RRSIG   DS 8 2 3600 ...
+~~~
 
-   ;; Query time: 226 msec
-   ;; SERVER: 69.36.157.30#53(69.36.157.30)
-   ;; WHEN: Wed Apr 15 13:34:43 AEST 2020
-   ;; MSG SIZE  rcvd: 500
+#  Updates to RFC 1034
 
-   %
-~~~
+   Replace
 
-   DNS responses sometimes contain optional data in the additional
-   section. Glue records however are not optional. Several other
-   protocol extensions, when used, are also not optional. This
-   includes TSIG [@RFC2845], OPT [@RFC6891], and SIG(0) [@RFC2931].
+   "Copy the NS RRs for the subzone into the authority section of the
+   reply.  Put whatever addresses are available into the additional
+   section, using glue RRs if the addresses are not available from
+   authoritative data or the cache.  Go to step 4."
+
+   with
+
+   "Copy the NS RRs for the subzone into the authority section of the
+   reply.  Put whatever addresses are available into the additional
+   section, using glue RRs if the addresses are not available from
+   authoritative data or the cache.  If all glue RRs do not fit, set TC=1 in
+   the header.  Go to step 4."
 
-##  Example two: Sibling Glue from the same delegating zone
+#  Sibling Glue
 
-   Sibling glue are glue records that are not contained in the delegating
-   zone itself, but in another delegated zone. In many cases, these are not
-   strictly required for resolution, since the resolver can make follow-on
-   queries to the same zone to resolve the nameserver addresses after
-   following the referral to the sibling zone. However, most nameserver
-   implementations provide them as an optimization to obviate the need
-   for extra traffic.
+   Sibling glue are glue records that are not contained in the delegated
+   zone itself, but in another delegated zone from the same parent. In many
+   cases, these are not strictly required for resolution, since the resolver
+   can make follow-on queries to the same zone to resolve the nameserver
+   addresses after following the referral to the sibling zone. However,
+   most nameserver implementations today provide them as an optimization
+   to obviate the need for extra traffic from iterative resolvers.
 
-   *COMMENT DW 20210715: This example would be better if it showed a
-   cyclic dependency.  As is the foo.test domain can be resolved because
-   bar.test doesn't have any other dependencies.*
+   This document clarifies that sibling glue (being part of all available
+   glue records) MUST be returned in referral responses, and that the
+   requirement to set TC=1 applies to sibling glue that cannot fit in the
+   response too.
+
+## Sibling Glue example
 
-~~~
 Here the delegating zone "test" contains 2 delegations for the
 subzones "bar.test" and "foo.test". The nameservers for "foo.test"
-consist of sibling glue for "bar.test" (ns1.bar.test and ns2.bar.test).
+require sibling glue from "bar.test" (ns1.bar.test and ns2.bar.test).
```

Comment:
I find 'The nameservers for "foo.test" require' to be confusing in a number of ways.  First I interpreted nameserver as the system running an authoritative server and hosting the foo.test zone.  But it really refers to the NS RRs.  Neither of those things require glue.  Its really the recursive name server that requires glue, no?

How about just deleting that sentence?  I think the requirements are already covered elsewhere.

Reply:
Ah, yes. I guess I could have said the foo.test "delegation" requires sibling glue. That's probably more accurate. But sure, I'm fine with just deleting this.